conans/client/graph/range_resolver.py

```diff
@@ -143,27 +143,24 @@ def _resolve_local(self, search_ref, version_range):
         if local_found:
             return self._resolve_version(version_range, local_found)
 
-    def _search_remotes(self, pattern, remotes):
-        remote = remotes.selected
-        if remote:
-            search_result = self._remote_manager.search_recipes(remote, pattern, ignorecase=False)
-            return search_result, remote.name
-
+    def _search_remotes(self, search_ref, remotes):
         for remote in remotes.values():
-            search_result = self._remote_manager.search_recipes(remote, pattern, ignorecase=False)
-            if search_result:
-                return search_result, remote.name
+            if not remotes.selected or remote == remotes.selected:
```

Comment:
This might change behavior. If a remote is specified explictly, but the reference is not found there, it will iterate to other remotes?

Reply:
Good catch! 🙌